.github/workflows/build.yml

```diff
@@ -19,20 +19,23 @@ jobs:
     steps:
       - name: Check out mitiq
         uses: actions/checkout@v4
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
           python-version: "3.12"
-      - name: Install Python dependencies
-        run: |
-          python -m pip install --upgrade pip
-          make install
+          
+      - name: Install the project
+        run: uv sync --all-extras
 
       - name: Check types with mypy
-        run: make check-types
+        run: uv run make check-types
```

Comment:
I like @bachase's suggestions of prefixing every command in the makefile with `uv run` so it doesn't have to change elsewhere. WDYT?